src/plugins/vis_types/vega/public/vega_view/vega_base_view.js

```diff
@@ -293,12 +293,8 @@ export class VegaBaseView {
   }
 
   updateVegaSize(view, dimensions) {
-    const width = Math.floor(
-      Math.max(0, dimensions?.width ?? this._container.getBoundingClientRect().width)
-    );
-    const height = Math.floor(
-      Math.max(0, dimensions?.height ?? this._container.getBoundingClientRect().height)
-    );
+    const width = Math.floor(Math.max(0, dimensions?.width ?? this._container.clientWidth));
+    const height = Math.floor(Math.max(0, dimensions?.height ?? this._container.clientHeight));
```

Comment:
I played a bit with the PR but I see the same issue (scrollbars visible) if you just resize the panel. The PR fixes when switching to full screen, but it doesn't work for the panel resize.
I'm not sure if that was also the previous behaviour

Reply:
Yeah, I've seen that before too where we are throttling the size update. I'll check with older versions and see if it's the same there.

Reply:
Ah, that's the reason? throttling? if so we just need to just to use a debounce strategy instead of throttling.

Reply:
Ok I switched the resize to debounce in fabb566.

### Throttle

![Zight Recording 2024-10-03 at 11 41 13 AM](https://github.com/user-attachments/assets/5f2f06c9-993e-47fe-90d9-787fa611983a)

### Debounce

![Zight Recording 2024-10-03 at 11 40 21 AM](https://github.com/user-attachments/assets/b59b4d9e-249b-47d0-babe-7e536d5e92fd)

Reply:
Removing debounce fixes this...

![Zight Recording 2024-10-07 at 12 49 39 PM](https://github.com/user-attachments/assets/a74af93a-1bd8-41f7-b6bd-d15d618e7bca)